pkg/objstore/s3/s3.go

```diff
@@ -379,6 +429,7 @@ func (b *Bucket) Iter(ctx context.Context, dir string, f func(string) error, opt
 }
 
 func (b *Bucket) getRange(ctx context.Context, name string, off, length int64) (io.ReadCloser, error) {
+	b.ReloadCredentials()
```

Comment:
Regarding this, are we calling this reloader function on each call? Won't this a little too expensive to have on each function call?